packages/um7/package.py

```diff
@@ -90,48 +78,147 @@ def _exe_name(self, opt_value):
+excl_dep                                           USE::mpl
+excl_dep                                           USE::mod_prism
+excl_dep                                           USE::mod_prism_proto
+excl_dep                                           USE::mod_prism_grids_writing
+excl_dep                                           USE::mod_prism_def_partition_proto
+excl_dep                                           USE::mod_prism_put_proto
+excl_dep                                           USE::mod_prism_get_proto
+excl_dep::script                                   EXE
+exe_dep                                            portio2a.o pio_data_conv.o pio_io_timer.o
+exe_name::flumeMain                                {EXE_NAME}
+pp                                                 1
+target                                             {EXE_NAME}
+tool::ar                                           ar
+tool::cc                                           mpicc
+tool::cflags                                       {FO} -g {FTRACEBACK} {FDEBUG} {FARCH} -fp-model precise
+tool::cpp                                          cpp
+tool::cppflags
+tool::cppkeys                                      {CPPKEYS}
+tool::fc                                           mpif90
+tool::fflags                                       {FO}  -g   -traceback  {FDEBUG} -V -i8 -r8      -fp-model precise {FFLAGS}
+tool::fflags::control::coupling::dump_received     {FO} {FG} {FTRACEBACK} {FDEBUG}            -mp1 -fp-model strict  {FFLAGS}
+tool::fflags::control::coupling::dump_sent         {FO} {FG} {FTRACEBACK} {FDEBUG}            -mp1 -fp-model strict  {FFLAGS}
+tool::fflags::control::coupling::oasis3_atmos_init {FO} {FG} {FTRACEBACK} {FDEBUG}    -i4 -r8 -mp1 -fp-model strict  {FFLAGS}
+tool::fflags::control::top_level::atm_step         -O0   -g  {FTRACEBACK} {FDEBUG} -V -i8 -r8 -mp1 -fp-model strict  {FFLAGS}
+tool::fflags::control::top_level::set_atm_pointers -O0   -g   -traceback  {FDEBUG}    -i8 -r8      -fp-model strict -ftz -std95
+tool::fflags::control::top_level::u_model          -O0   -g  {FTRACEBACK} {FDEBUG} -V -i8 -r8 -mp1 -fp-model strict  {FFLAGS}
+tool::fpp                                          cpp
+tool::fppflags                                     -P -traditional
+tool::fppkeys                                      {CPPKEYS}
+
+tool::geninterface                                 none
+tool::ld                                           mpif90
+tool::ldflags                                      {FOBLANK} -g -traceback {FDEBUG} -static-intel {libs}
+        """
+        bld_cfg_path = self._bld_cfg_path()
+        with open(bld_cfg_path, "w") as bld_cfg_file:
+            bld_cfg_file.write(config)
 
+
+    def build(self, spec, prefix):
+        """
+        Use FCM and the Intel compiler to build the executable.
+        """
         if self.compiler.name == "intel":
             env["platform_config_dir"] = "nci-x86-ifort"
```

Comment:
Now that we create the config file from the SPD, can we define config dir whatever we like?

Reply:
Removed the check since the environment variable is not used.